src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -749,6 +749,82 @@ private static String captureSystemStream(Runnable runnable, PrintStream stream,
         return out.toString();
     }
 
+    @Test(expectedExceptions = CommandLineException.CommandLineParserInternalException.class)
+    public void testWithBoundariesArgumentsForNoNumeric() {
+        @CommandLineProgramProperties(summary = "broken tool",
+                oneLineSummary = "broken tool",
+                programGroup = TestProgramGroup.class)
+        class WithBoundariesArgumentsForString {
+            @Argument(doc = "String argument", optional = true, minValue = 0, maxValue = 30)
+            public String stringArg = "string";
+        }
+        final CommandLineArgumentParser clp = new CommandLineArgumentParser(new WithBoundariesArgumentsForString());
+        clp.parseArguments(System.err, new String[]{});
+    }
+
+    @CommandLineProgramProperties(
+            summary = "tool with boundaries",
+            oneLineSummary = "tools with boundaries",
+            programGroup = TestProgramGroup.class
+    )
+    public class WithBoundariesArguments {
+
+        @Argument(doc = "Double argument in the range [10, 20]", optional = true, minValue = 10, maxValue = 20)
+        public Double doubleArg = 15d;
+
+        @Argument(doc = "Integer in the range [0, 30]", optional = true, minValue = 0, maxValue = 30)
+        public int integerArg = 20;
+
+        @Argument(doc = "String argument", optional = true)
```

Comment:
I maintain it for checking if it does not blows up if the argument is a `String`, but it's true that it is redundant with other tests already implemented. Removed!